crates/circuit/src/dag_circuit.rs

```diff
@@ -4957,7 +4955,7 @@ impl DAGCircuit {
     pub fn collect_runs(
         &self,
         namelist: HashSet<String>,
-    ) -> Option<impl Iterator<Item = Vec<NodeIndex>> + '_> {
+    ) -> PyResult<impl Iterator<Item = Vec<NodeIndex>> + '_> {
```

Comment:
```suggestion
    ) -> impl Iterator<Item = Vec<NodeIndex>> {
```

This method shouldn't need a result type after the comment below has been addressed.